src/Compilers/Core/Portable/MetadataReader/MetadataHelpers.cs

```diff
@@ -756,6 +763,12 @@ internal static string UnmangleMetadataNameForArity(string emittedTypeName, int
                         else
                         {
                             // This is a new child namespace
+                            if (cmp > 0)
+                            {
+                                // The sort order is violated for child namespace names. Obfuscation is the likely reason for this. 
```

Comment:
The doc comment for typesByNS documents expected sort order for groups. For conventional namespace names, nested namespace names will be in the same sort order.